analyses/infercnv-consensus-cell-type/template-notebooks/SCPCP000015_explore-infercnv-results.Rmd

```diff
@@ -0,0 +1,418 @@
+* `tumor-immune`: Cells labeled as tumor by cell-type-ewings, but immune by `cell-type-consensus`
+* `tumor`: Cells labeled as tumor by `cell-type-ewings`
+* `endo`: An endothelial cell type
+* `immune`: An immune cell type
+* If none of the above, their existing `cell-type-ewings` annotation
+
+Ideally, we'll see that tumor and Unknown cells from will have relatively higher CNV scores compared to other cell types, in particular those cell types in the reference.
+
+```{r fig.height=12, message=FALSE, warning=FALSE}
+# drop cells from other libraries
+query_has_cnv_df <- has_cnv_df |>
+  dplyr::filter(library_id == params$library_id)
+
+ggplot(query_has_cnv_df) +
+  aes(
+    y = celltype,
+    x = total_cnv_per_cell,
+    fill = cell_group
+  ) +
+  ggridges::geom_density_ridges2(alpha = 0.7, scale = 0.85)
+```
+
+
+## UMAPs
+
+Next, we'll visualize the total CNV counts as a UMAP.
+Below, the UMAP colored by cell type is shown on the left, and the UMAP colored by total CNV counts is shown on the right.
+Ideally, unknown and tumor cells will overlap with regions in the UMAP with higher total CNV.
+
+
+```{r, fig.height = 10}
+celltype_umap <- ggplot(umap_df) +
+  aes(x = UMAP1, y = UMAP2, color = celltype) +
+  geom_point(size = 0.1, alpha = 0.3) +
+  guides(color = guide_legend(override.aes = list(size = 2, alpha = 1))) +
+  labs(title = "Cell types") +
+  umap_theme
+
+cnv_umap <- ggplot(query_has_cnv_df) +
+  aes(x = UMAP1, y = UMAP2, color = total_cnv_per_cell) +
+  geom_point(size = 0.1, alpha = 0.3) +
+  scale_color_viridis_c() +
+  labs(title = "Total CNV per cell") +
+  umap_theme
+
+celltype_umap / cnv_umap
+```
+
+
+## Clusters
+
+Part of `inferCNV`'s algorithm involves clustering the data, and poor clusters have the potential to lead to inaccurate inferences of CNV.
+To see whether we'd like to further tune the clustering parameters, we'll get a sense of their size and rough composition here.
+
+How many clusters are there, and how big are they?
+We'll show this separately for reference vs query cells.
+
+```{r}
+cluster_count_df <- has_cnv_df |>
+  dplyr::count(subcluster, cell_group)
+ggplot(cluster_count_df) +
+  aes(x = n, fill = cell_group) +
+  geom_histogram(color = "grey20", bins = 15) +
+  facet_wrap(vars(cell_group), scales = "free")
+```
+
+Is there any relationship between cluster size and the number of CNVs?
+We would ideally not want there to be a relationship since this might indicate a bias.
+
+```{r}
+cluster_size_df <- has_cnv_df |>
+  dplyr::add_count(subcluster, name = "cluster_size") |>
+  dplyr::select(
+    total_cnv_per_cell,
+    cluster_size
+  ) |>
+  dplyr::distinct()
+
+ggplot(cluster_size_df) +
+  aes(x = cluster_size, y = total_cnv_per_cell) +
+  geom_point() +
+  geom_smooth(method = "lm")
+```
+
+
+Let's also look more specifically at the `reference` clusters, where we might expect some batch effects since cells were taken from several different libraries.
```

Comment:
In the older version, you had a plot that looked at the distribution of cell types in each of the query clusters too. Can you add that back? I would expect any normal cells that are not part of the reference should cluster together and be separate from tumor cells.